examples/with_plugs.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Example OpenHTF test logic.
+
+Run with (your virtualenv must be activated first):
+
+  python with_plugs.py
+
+Afterwards, take a look at the hello_world.json output file.  This will
+give you a basic idea of what a minimal test outputs.
+
+For more information on measurements, see the measurements.py example.
+
+TODO(someone): Write an output example.
+For more information on output, see the output.py example.
+
+with_plugs() is most useful when you have a test that has to happen in the same
+(or mostly similar) ways across multiple interfaces, such testing connectivity
+on a quad-NIC board. Instead of creating 4 phases with 4 plugs, you create 1 phase
+with 4 subclasses of the same plug and use with_plugs() to create the 4 phases.
+"""
+import itertools
+import subprocess
+import time
+
+import openhtf as htf
+from openhtf import plugs
+
+
+class PingPlug(plugs.BasePlug):
+  """This plug simply does a ping against the host attribute."""
+  enable_remote = True
+  host = None
+
+  def __init__(self):
+    # This should only ever be constructed by a subclass, so this is only
+    # checking that the subclass set host correctly.
+    assert self.host is not None
+
+  def _get_command(self, count):
+    # Returns the commandline for pinging the host.
+    return [
+      'ping',
+      '-c',
+      str(count),
+      self.host,
+    ]
+
+  def run(self, count):
+    command = self._get_command(count)
+    print "running: %s" % command
```

Comment:
probably `''.join(command)` instead of just command? it's a little odd to see a list in the context of "running"